vk/middleware.go

```diff
@@ -49,50 +78,32 @@ func enableCors(ctx *Ctx, domain string) {
 	}
 }
 
-// generate a HandlerFunc that passes the request through a set of Middleware first and Afterware after
-func augmentHttpHandler(inner HandlerFunc, middleware []Middleware, afterware []Afterware) HandlerFunc {
-	return func(r *http.Request, ctx *Ctx) (interface{}, error) {
-		defer func() {
-			// run the afterware (which cannot affect the response)
-			// even if something in the request chain fails
-			for _, a := range afterware {
-				a(r, ctx)
+// ErrorMiddleware returns a middleware that wraps a handler.
+func ErrorMiddleware() Middleware {
```

Comment:
Does it make sense to inject the logger as an argument to the handler closure? 

I understand we pass the logger around in the context object today but we could use this opportunity to fix forward even if we keep a logger reference in the context object for compatibility reasons.

Reply:
Perhaps in another PR but it may make sense to create an ErrorMapper interface which can convert an application error into an HttpError which includes a status along with a message.

Reply:
Yes, the logger will be passed in later, but in this PR the ctx is not touched. See the "out of scope for this PR -> logs" section of the PR message 🙂